shell/platform/android/io/flutter/view/TextureRegistry.java

```diff
@@ -89,6 +95,35 @@ interface SurfaceProducer extends TextureEntry {
      */
     Surface getSurface();
 
+    /**
+     * Sets a callback that is notified when
+     *
+     * @param callback The callback to notify, or null to remove the callback.
+     */
+    void setOnSurfaceRecreatedCallback(OnSurfaceRecreatedCallback callback);
+
+    /** Callback invoked by {@link #setOnSurfaceRecreatedCallback(OnSurfaceRecreatedCallback)}. */
+    interface OnSurfaceRecreatedCallback {
+      /**
+       * Invoked when a new surface was forcibly recreated by {@link SurfaceProducer}.
+       *
+       * <p>Typically plugins will use this callback as a signal to redraw.
+       */
+      void onSurfaceRecreated();
```

Comment:
We need to notify of destroy so that they stop rendering into it.

I would copy the Android SurfaceHolder.Callback interface design